articles/django-react-app/index.md

```diff
@@ -0,0 +1,511 @@
+
+We use the built-in JavaScript `map` functionality in our JavaScript XML (JSX). It allow us to iterate over the list of items and display them. We use the curly braces to encapsulate our item property since it evaluate Javascript expressions during compilation.
+
+When the component file is called it calls the `render()` method by default displaying th JSX syntax.
+
+We use [arrow functions](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions) because they shorten our functions declaration.
+
+Your User Interface should resemble the one below:
+
+![frontend](dummy_data.jpg)
+
+It is time to consume the API we created earlier.
+
+```bash
+cd backend
+python manage.py runserver
+```
+
+We will need to modify the `frontend/package.json`:
+
+```javascript
+// frontend/package.json
+
+[...]
+"name": "frontend",
+  "version": "0.1.0",
+  "private": true,
+  "proxy": "http://localhost:8000",   // Add this
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.11.9",
+    "@testing-library/react": "^11.2.3",
+    "@testing-library/user-event": "^12.6.0",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-scripts": "4.0.1",
+    "web-vitals": "^0.2.4"
+  },
+  [...]
+```
+
+First, lets install the bootstrap module that we will use to create a jumbotron header:
+
+```bash
+npm install --save react-bootstrap-validation
+```
+
+We then import the `Jumbotron` in our `App.js` file.
+
+To consume our API instead of the artificial data, update the `frontend/src/App.js` with the snippet below:
+
+```javascript
+// frontend/src/App.js
+
+import React, { Component } from "react"
+import Jumbotron from 'react-bootstrap/Jumbotron' // Add this
+
+
+class App extends Component {
+    state = {
+      todoList: []
+    };
+
+    async componentDidMount() {
+      try {
+        const res = await fetch('http://localhost:8000/api/todos/');
+        const todoList = await res.json();
+        this.setState({
+          todoList
+        });
+      } catch (e) {
+        console.log(e);
+    }
+    }
+
+    render() {
+      return (
+        <div>
+          <Jumbotron>
+          <h1>Todo App</h1>
+          </Jumbotron>
+          {this.state.todoList.map(item => (
+            <div>
+              <h1>{item.title}</h1>
+              <span>{item.description}</span>
+            </div>
+          ))}
+        </div>
+      )
+    }
+  }
+  
+export default App;
+```
+
+We assign our `todoList` an empty array because we are going to fetch our data from an API.
+
+First, we wrap Fetch in a `try/catch` block to handle any network errors. We then call `fetch()` with the `await` keyword, where we pass our apiEndpoints. We're telling the  `async` function to stop executing untill the promise is resolved at which point it can resume execution and return the resolved value. Rather than getting promises, we will get back the parsed JSON data that we expect.
```

Comment:
```suggestion
First, we wrap `fetch()` in a `try/catch` block to handle any network errors. We then call `fetch()` with the `await` keyword, where we pass our API endpoints. We're telling the `async` function to stop executing until the promise is resolved at which point it'll resume execution and return the resolved value. Rather than getting promises, we will get back the parsed JSON data that we expect.
```